server/resources.py

```diff
@@ -1,34 +1,38 @@
-from flask_restful import Resource, reqparse
-from models.users import UserModel
-
-# TODO: Replace with Marshmallow
-parser = reqparse.RequestParser()
-parser.add_argument('username', help = 'This field cannot be blank', required=True)
-parser.add_argument('password', help = 'This field cannot be blank', required=True)
+from models.users import UserModel, user_schema, users_schema
+from flask_restful import Resource
+from flask import request
 
 class UserRegistration(Resource):
     def post(self):
-        data = parser.parse_args()
+        data = user_schema.load(request.json)
+
+        # Check if email already exists
+        if UserModel.find_by_email(data['email']):
+            return {'message': 'User {} already exists'.format(data['email'])}, 409
+
         new_user = UserModel(
-            username = data['username'],
-            password = data['password']
+            email = data['email'],
+            password = UserModel.generate_hash(data['password'])
         )
         try:
             new_user.save_to_db()
-            return {
-                'message': 'User {} was created'.format(data['username'])
-            }
+            return {'message': 'User {} was created'.format(data['email'])}, 201
         except Exception as e:
             return {'message': 'Something went wrong'}, 500
 
-        # TODO: Need to raise 409 error if username already exists
-
 
 
 class UserLogin(Resource):
     def post(self):
-        data = parser.parse_args()
-        return data
+        data = user_schema.load(request.json)
+        current_user = UserModel.find_by_email(data['email'])
+        if not current_user:
+            return {'message': 'User {} doesn\'t exist'.format(data['email'])}
```

Comment:
It's a security issue to tell the user that the user doesn't exist in logging in routes and can be exploited. Just always return Wrong credentials